lib/mp4/transmuxer.js

```diff
@@ -891,17 +892,10 @@ CoalesceStream.prototype.flush = function(flushSource) {
 
   this.emittedTracks += this.pendingTracks.length;
 
-  initSegment = mp4.initSegment(this.pendingTracks);
+  event.serializedTracks = mp4Utils.serializeTracks(this.pendingTracks);
```

Comment:
There are two options here.
1. Instead of manually serializing the whole track, you can simply convert the `sps` and `pps` into arrays of arrays and then convert them back to arrays of typed-arrays on the other size of the worker.
1. The entries in the `sps` and `pps` can be duplicated in the worker and sent via a zero-copy out of the worker much like the `data` is..

Either way, I don't think the muxer is the place to do the conversion since it shouldn't know or care if it is in a typed array or not. The place for this kind of `data-marshalling` is in the `transmux_worker.js` file in `videojs-contrib-media-sources`.